modules/cloudsql-instance/variables.tf

```diff
@@ -195,6 +196,15 @@ variable "project_id" {
   type        = string
 }
 
+variable "psc_config" {
+  description = "Whether Private Service Connect (PSC) connectivity is enabled for this instance and list of consumer projects that are allow-listed for PSC connections to this instance."
+  type = object({
+    allowed_consumer_projects = optional(list(string), [])
+    psc_enabled               = optional(bool, true)
```

Comment:
https://cloud.google.com/sql/docs/mysql/configure-private-service-connect#create-cloud-sql-instance-psc-enabled
Following this doc, _consumer projects_ I think that an empty value means that no project has access.
I think it's better to leave the variable as it is because you can decide, for some reason, to remove a project and in the future add a new one, without disabing the PSC.